contexts/pathogen-context/filter-update-steps/apply-new-selected-filters.tsx

```diff
@@ -44,9 +44,43 @@ export function filterData(
           // Check for any overlap in the sampling period
           return itemEndDate >= filterStartDate;
         }
+        case "samplingStartDate": {
```

Comment:
I don't like this very much either. I think this function could use a bit of love but this PR is large enough as-is so I'm leaving it alone for now.